ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/util/DataColumnSidecarELRecoveryManagerImplTest.java

```diff
@@ -0,0 +1,233 @@
+    dataColumnSidecarELRecoveryManagerCustom.onNewBlock(block, Optional.empty());
+    assertThat(asyncRunner.hasDelayedActions()).isFalse();
+    verifyNoInteractions(executionLayer);
+
+    final int fuluSlot = minimalWithFuluForkEpoch.slotsPerEpoch(UInt64.ZERO) + 1;
+    final SignedBeaconBlock block2 = dataStructureUtil.randomSignedBeaconBlock(fuluSlot);
+    dataColumnSidecarELRecoveryManagerCustom.onSlot(UInt64.valueOf(fuluSlot));
+    dataColumnSidecarELRecoveryManagerCustom.onNewBlock(block2, Optional.empty());
+    assertThat(asyncRunner.hasDelayedActions()).isTrue();
+    asyncRunner.executeQueuedActions();
+    verify(executionLayer).engineGetBlobs(any(), any());
+  }
+
+  @Test
+  public void onNewBlock_ignoresLocal() {
+    final SignedBeaconBlock block =
+        dataStructureUtil.randomSignedBeaconBlock(currentSlot.longValue());
+    dataColumnSidecarELRecoveryManager.onSlot(currentSlot);
+    dataColumnSidecarELRecoveryManager.onNewBlock(block, Optional.of(RemoteOrigin.LOCAL_PROPOSAL));
+    assertThat(asyncRunner.hasDelayedActions()).isFalse();
+    verifyNoInteractions(executionLayer);
+  }
+
+  @Test
+  public void onNewDataColumnSidecar_ignoresLocalOrRecovered() {
+    final SignedBeaconBlock block =
+        dataStructureUtil.randomSignedBeaconBlock(currentSlot.longValue());
+    final DataColumnSidecar dataColumnSidecar =
+        dataStructureUtil.randomDataColumnSidecarWithInclusionProof(block, UInt64.ZERO);
+    dataColumnSidecarELRecoveryManager.onSlot(currentSlot);
+    dataColumnSidecarELRecoveryManager.onNewDataColumnSidecar(
+        dataColumnSidecar, RemoteOrigin.LOCAL_PROPOSAL);
+    dataColumnSidecarELRecoveryManager.onNewDataColumnSidecar(
+        dataColumnSidecar, RemoteOrigin.RECOVERED);
+    assertThat(asyncRunner.hasDelayedActions()).isFalse();
+    verifyNoInteractions(executionLayer);
+  }
+
+  @Test
+  public void onNewDataColumnSidecar_startsRecovery() {
+    final SignedBeaconBlock block =
+        dataStructureUtil.randomSignedBeaconBlock(currentSlot.longValue());
+    final DataColumnSidecar dataColumnSidecar =
+        dataStructureUtil.randomDataColumnSidecarWithInclusionProof(block, UInt64.ZERO);
+    dataColumnSidecarELRecoveryManager.onSlot(currentSlot);
+    dataColumnSidecarELRecoveryManager.onNewDataColumnSidecar(
+        dataColumnSidecar, RemoteOrigin.GOSSIP);
+    assertThat(asyncRunner.hasDelayedActions()).isTrue();
+    asyncRunner.executeQueuedActions();
+    verify(executionLayer).engineGetBlobs(any(), any());
+  }
+
+  @Test
+  public void shouldNotPublish_whenNotCurrentSlot() {
+    final SignedBeaconBlock block =
+        dataStructureUtil.randomSignedBeaconBlock(currentSlot.longValue());
+    dataColumnSidecarELRecoveryManager.onSlot(currentSlot.minus(1));
+    dataColumnSidecarELRecoveryManager.onNewBlock(block, Optional.empty());
+    assertThat(asyncRunner.hasDelayedActions()).isFalse();
+    verifyNoInteractions(executionLayer);
```

Comment:
Does it make sense to `verifyNoInteraction` here when we have not called `asyncRunner.executeQueuedActions();`? 

Given we check that `asyncRunner.hasDelayedActions()` is false, we do not need to `verifyNoInteractions`.

Reply:
Doesn't make a lot sense but but 
1. I'm paranoid
2. It's good to write tests not looking at the implementation, because it could be changed with time and other reasons